misc/factorial_recursive.c

```diff
@@ -0,0 +1,85 @@
+/**
+ * @file
+ * @brief [Factorial](https://en.wikipedia.org/wiki/Factorial) with recursive function call
+ * [Recurrence relation](https://en.wikipedia.org/wiki/Recurrence_relation)
+ * @details
+ * This version of the factorial computation function obtains the result by
+ * recursively calling the function
+ * @author [Nicola Masarone](https://github.com/nicola-masarone)
+ *
+ */
+
+#include <assert.h>  /// allows the use of the assert() macro for a diagnostic test of the function
+#include <stdio.h>  /// allows the use of standard input output functions such as scanf() and printf()
+
+/**
+ * Function documentation
+ * This function calculates the factorial of a number: n! = n*(n-1)*...*1 
+ * From its definition we can calculate the factorial as: n! = n*(n-1)! 
+ * that is, with recursive call of the function applied to the argument decreased by 1. 
+ * The recursion must stop when it reaches 0! which by definition is equal to 1.
+ * @param val input value to calculate the factorial of
+ * @returns factorial of the input value
+ */
+long long factorial_recursive(int val)
```

Comment:
```suggestion
long long factorial_recursive(const int& val)
```

Reply:
Don't understand this suggested change. Is there any _real need_ to use **const int&** instead of just **int** ?
I supposed this code should be useful for beginners too and I think it's much more readable in this way. 
I would prefer to leave it just as it is.
Thanks again for your suggestions.

Reply:
This parameter (`val`) is not being modified, it is just being used.
It's better to make it a `const` and pass it by reference using `&`.

If the value **is** being modified, we shall not use `const`, otherwise it'll throw an error.
[Reference](https://www.quora.com/What-is-const-in-C-programming).